par_test.go

```diff
@@ -314,87 +313,143 @@ func BenchmarkReduce(b *testing.B) {
 	})
 }
 
-var result bool
-
 func BenchmarkAny(b *testing.B) {
-	rand.Seed(time.Now().Unix())
+	rand.Seed(1)
 	collections := CreateCollections(10000)
-	maxSum := CollectionsGetMaxSum(collections)
 
-	b.Run("serial", func(b *testing.B) {
-		var r bool
+	b.Run("serial with match", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
+			needle := collections[len(collections)*n/b.N].NumbersSum()
+			var result bool
 			for _, c := range collections {
-				if c.NumbersSum() == maxSum {
-					r = true
+				if c.NumbersSum() == needle {
+					result = true
 					break
 				}
 			}
+			_ = result
 		}
-		result = r
 	})
-	b.Run("parallel", func(b *testing.B) {
-		var r bool
+	b.Run("parallel with match", func(b *testing.B) {
 		for n := 0; n < b.N; n++ {
-			r = par.Any(collections, func(c Collection) bool {
-				return c.NumbersSum() == maxSum
+			needle := collections[len(collections)*n/b.N].NumbersSum()
```

Comment:
I would recommend using reset to avoid this overhead in benchmark, similarly below.
```suggestion
			needle := collections[len(collections)*n/b.N].NumbersSum()
			b.ResetTimer()
```

Reply:
I think doing a `b.ResetTimer()` inside the `n < b.N` loop would defeat the purpose 🤔 in any case, this computation happens for both the serial reference implementation and the parallel implementation, so the results should remain comparable with each other due to amortization, while not being representative by themselves. But I don't think micro-benchmark results are really ever representative without a comparison to a reference implementation in the same run, so that's not really an issue.

Reply:
you are right I missed the context there.